mule-extensions-api/src/main/java/org/mule/runtime/extension/api/ExtensionConstants.java

```diff
@@ -385,5 +391,21 @@ public final class ExtensionConstants {
    */
   public static final String VERSION_PROPERTY_NAME = "version";
 
+  /**
+   * The Java versions that an Extension is assumed to support if not explicitly stated
+   *
+   * @since 1.5.0
+   */
+  public static final Set<String> DEFAULT_SUPPORTED_JAVA_VERSIONS =
+      unmodifiableSet(new LinkedHashSet<>(asList(JAVA_VERSION_8, JAVA_VERSION_11)));
+
+  /**
+   * The full set of Java versions that an Extension can possibly support
+   *
+   * @since 1.5.0
+   */
+  public static final Set<String> ALL_SUPPORTED_JAVA_VERSIONS =
+      unmodifiableSet(new LinkedHashSet<>(asList(JAVA_VERSION_8, JAVA_VERSION_11, JAVA_VERSION_17)));
+
```

Comment:
Is the idea to update this on each release/continuously?

Reply:
yes, but not continusly as we'll add/remove java versions once every (hopefully) 30 months